### PR TITLE
add resource azurerm_cosmosdb_sql_trigger

### PR DESCRIPTION
**Pull request by ms-henglu**
The added tests are listed as follows.
```
=== RUN   TestAccCosmosDbSQLTrigger_basic
=== PAUSE TestAccCosmosDbSQLTrigger_basic
=== CONT  TestAccCosmosDbSQLTrigger_basic
--- PASS: TestAccCosmosDbSQLTrigger_basic (1271.04s)
=== RUN   TestAccCosmosDbSQLTrigger_requiresImport
=== PAUSE TestAccCosmosDbSQLTrigger_requiresImport
=== CONT  TestAccCosmosDbSQLTrigger_requiresImport
--- PASS: TestAccCosmosDbSQLTrigger_requiresImport (1224.34s)
=== RUN   TestAccCosmosDbSQLTrigger_update
=== PAUSE TestAccCosmosDbSQLTrigger_update
=== CONT  TestAccCosmosDbSQLTrigger_update
--- PASS: TestAccCosmosDbSQLTrigger_update (1339.53s)
```

### CODE REVIEW

File: azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosEntityName,
+			},
+
+			"database_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosEntityName,
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go:68]: we can parse this out of a resource id?

File: azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+			},
+
+			"body": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"trigger_operation": {
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go:77]: trigger is in the resource name?
```suggestion
			"operation": {
```

File: azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+				ValidateFunc: validation.StringInSlice([]string{
+					string(documentdb.All),
+					string(documentdb.Create),
+					string(documentdb.Update),
+					string(documentdb.Delete),
+					string(documentdb.Replace),
+				}, false),
+			},
+
+			"trigger_type": {
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go:89]:
```suggestion
			"type": {
```

**Comment by ms-henglu** [comment]
@katbyte , hi, I updated this PR, could you please take another look? Thanks!

File: azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+	containerId, _ := parse.SqlContainerID(d.Get("container_id").(string))
+	resourceGroup := containerId.ResourceGroup
+	accountName := containerId.DatabaseAccountName
+	containerName := containerId.ContainerName
+	databaseName := containerId.SqlDatabaseName
+	body := d.Get("body").(string)
+	triggerOperation := d.Get("operation").(string)
+	triggerType := d.Get("type").(string)
+
+	id := parse.NewSqlTriggerID(subscriptionId, resourceGroup, accountName, databaseName, containerName, name)
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go:99]: we don't need to assign these to an intermediate?
```suggestion

	body := d.Get("body").(string)
	triggerOperation := d.Get("operation").(string)
	triggerType := d.Get("type").(string)

	id := parse.NewSqlTriggerID(subscriptionId, containerId.ResourceGroup, containerId.AccountName, containerId.DatabaseName, containerId.Name, name)
```

File: azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+			Resource: &documentdb.SQLTriggerResource{
+				ID:               &name,
+				Body:             &body,
+				TriggerType:      documentdb.TriggerType(triggerType),
+				TriggerOperation: documentdb.TriggerOperation(triggerOperation),
+			},
+			Options: &documentdb.CreateUpdateOptions{},
+		},
+	}
+	future, err := client.CreateUpdateSQLTrigger(ctx, resourceGroup, accountName, databaseName, containerName, name, createUpdateSqlTriggerParameters)
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go:124]: same here

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @ms-henglu - one last comment and this should be good to merge

**Comment by ms-henglu** [comment]
@katbyte , thanks for code review, I updated pr according to your suggestion.

**Review by katbyte — APPROVED** [review]
Thanks @ms-henglu - LGTM 👍

**Pull request merged**